Makefile

```diff
@@ -0,0 +1,36 @@
+GO ?= go
+GH_ORG ?= Azure-Samples
+ROOT = github.com/$(GH_ORG)/azure-sdk-for-go-samples
+BASE = $(GOPATH)/src/$(ROOT)
+
+# tested on 1.9+, no need to exclude /vendor
+# to specify packages to skip: `PKGS_SKIP_RE='storage|graphrbac' make ...`
+PKGS_SKIP_RE ?= ''
+PKGS         != $(GO) list ./... | sed 's@$(ROOT)/@./@' | grep -v -E "$(PKGS_SKIP_RE)"
+
+# uses Azure resources
+test: dep
+	$(GO) test -v $(PKGS)
+
+test_pr: lint
+	$(GO) build -v \
+		$(shell $(GO) list ./... | sed 's@$(ROOT)/@./@' | grep -v -E "$(PKGS_SKIP_RE)")
+
+dep:
+	$(GO) get -u github.com/golang/dep/cmd/dep
+	cd $(BASE) && dep ensure
+
+lint: dep
+	$(GO) get -v github.com/alecthomas/gometalinter
+	gometalinter --install
+	# TODO: fix problems and enable all tests
+	# TODO: address warnings
+	gometalinter --errors \
+		--enable=gofmt \
+		--enable=goimports \
+		--disable=vet \
```

Comment:
Interesting, why no `vet`? It has caught a mistake or two from my-end where I get a closure wrong with parallelism. :)

Reply:
It indeed catches many issues here :). I'd like to create a task to address issues and re-enable `vet`, just not able to do that at the moment.